packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts

```diff
@@ -213,6 +213,10 @@ export const useFieldState = <
       tempValueStrAndroid: null,
     }));
 
+    if (valueManager.areValuesEqual(utils, state.value, value)) {
```

Comment:
Before, we were not even calling `publishValue`.
But it's problematic with `Backspace` because the state was not updated when pressing `Backspace` on an already empty section, causing the selection to be removed by the browser.
So now we always update the state but we don't publish `onChange` if not needed